tangelo/problem_decomposition/dmet/_helpers/dmet_scf_guess.py

```diff
@@ -55,3 +52,49 @@ def dmet_fragment_guess(t_list, bath_orb, chemical_potential, norb_high, number_
     frag_guess = np.dot(eigenvectors[ :, : int(number_active_electron/2)], eigenvectors[ :, : int(number_active_electron/2)].T) * 2
 
     return frag_guess
+
+
+def dmet_fragment_guess_rohf_uhf(bath_orb, chemical_potential, norb_high,
+    number_active_electron, active_fock_alpha, active_fock_beta, n_active_alpha,
+     n_active_beta):
+    """Construct the guess orbitals.
+
+    Args:
+        bath_orb (numpy.array): The bath orbitals (float64).
+        chemical_potential (float64): The chemical potential.
+        norb_high (int): The number of orbitals in the fragment calculation.
+        number_active_electron (int): The number of electrons in the fragment
+            calculation.
+        active_fock_alpha (numpy.array): The fock matrix from the low-level
+            calculation for the alpha electrons (float64).
+        active_fock_beta (numpy.array): The fock matrix from the low-level
+            calculation for the beta electrons (float64).
+        n_active_alpha (int): The number octive alpha electrons.
+        n_active_beta (int): The number octive beta electrons.
+
+    Returns:
+        frag_guess (numpy.array): The guess orbitals (float64).
+    """
+
+    n_spin = n_active_alpha - n_active_beta
+    n_pair = (number_active_electron - n_spin) // 2
+    new = {"alpha": n_pair + n_spin, "beta": n_pair}
+    active_fock = {"alpha": active_fock_alpha, "beta": active_fock_beta}
+    frag_guess = dict()
+
+    for spin in {"alpha", "beta"}:
+        # Construct the fock matrix of the fragment (subtract the chemical potential for consistency)
+        fock_fragment = bath_orb[:, : norb_high].T @ active_fock[spin] @ bath_orb[:, : norb_high]
+        potential_diag = np.zeros_like(fock_fragment)
+        np.fill_diagonal(potential_diag, chemical_potential)
+        fock_fragment -= potential_diag
+
+        # Diagonalize the fock matrix and get the eigenvectors
+        eigenvalues, eigenvectors = scipy.linalg.eigh(fock_fragment)
+        eigenvectors = eigenvectors[:, eigenvalues.argsort()]
+
+        # Extract the eigenvectors of the occupied orbitals as the guess orbitals
+        # Introduce alpha- and beta-electrons
+        frag_guess[spin] = np.dot(eigenvectors[:, :int(new[spin])], eigenvectors[:, :int(new[spin])].T)
+
+    return np.array((frag_guess["alpha"], frag_guess["beta"])), [new["alpha"], new["beta"]]
```

Comment:
I cannot parse this line. The docs claims a numpy array of float64.

I don't know what `np.array((frag_guess["alpha"], frag_guess["beta"]))` returns, but you return whatever that is as well as a list of two numpy arrays (`[new["alpha"], new["beta"]]`). I don't know what's going on.